crates/services/parallel-executor/src/scheduler.rs

```diff
@@ -0,0 +1,1179 @@
+                    continue 'outer;
+                }
+
+                self.execute_batch(
+                    consensus_parameters_version,
+                    components,
+                    batch,
+                    nb_batch_created,
+                    nb_transactions,
+                    storage_with_da.clone(),
+                )?;
+
+                nb_batch_created += 1;
+                nb_transactions += batch_len;
+            } else if self.current_execution_tasks.is_empty() {
+                tokio::select! {
+                    _ = new_tx_notifier.notified() => {
+                        self.new_executable_transactions();
+                    }
+                    _ = tokio::time::sleep_until(deadline) => {
+                        break 'outer;
+                    }
+                }
+            } else {
+                tokio::select! {
+                    _ = new_tx_notifier.notified() => {
+                        self.new_executable_transactions();
+                    }
+                    result = self.current_execution_tasks.select_next_some() => {
+                        match result {
+                            Ok(res) => {
+                                match res {
+                                    Ok(res) => {
+                                        if !res.skipped_tx.is_empty() {
+                                    self.sequential_fallback(block_height, &consensus_parameters, res.batch_id, res.txs, res.coins_used, res.coins_created).await?;
+                                    continue;
+                                }
+                                self.register_execution_result(res);
+                                    },
+                                    Err(e) => {
+                                        return Err(SchedulerError::ExecutionError(e));
+                                    }
+                                }
+
+                            }
+                            _ => {
+                                return Err(SchedulerError::InternalError(
+                                    "Worker execution failed".to_string(),
+                                ));
+                            }
+                        }
+                    }
+                    _ = tokio::time::sleep_until(deadline) => {
+                        break 'outer;
+                    }
+                }
+            }
+        }
+
+        self.wait_all_execution_tasks(
+            block_height,
+            &consensus_parameters,
+            block_constraints.total_execution_time,
+        )
+        .await?;
+
+        let mut res = self.verify_coherency_and_merge_results(
+            nb_batch_created,
+            components.header_to_produce,
+            l1_execution_data,
+        )?;
+
+        if self.blob_transactions.is_empty() {
+            let mut merged: Changes = Changes::default();
+            for changes in res.changes.extract_list_of_changes() {
```

Comment:
Hmm, merging everything together will increase execution time in the case of blobs. Do we really need to merge everything? Maybe we can revisit the implementation for `InMemoryStorage` with the list of `Changes`? Maybe it is already good enough and will not fail, or do double spend?

Reply:
I didn't found any simple solutions to do it so I dropped it.